test/dune

```diff
@@ -1,23 +1,6 @@
-(test
- (name pure_html_test)
- (package pure-html)
- (libraries pure_html)
- (action
-  (diff expected.html got.html)))
-
-(rule
- (with-stdout-to
-  got.html
-  (run ./pure_html_test.exe)))
-
-(test
+(library
  (name dream_html_test)
- (package dream-html)
- (libraries dream_html)
- (action
-  (diff expected.txt got.txt)))
-
-(rule
- (with-stdout-to
-  got.txt
-  (run ./dream_html_test.exe)))
+ (inline_tests)
+ (libraries dream_html pure_html fmt)
+ (preprocess
+  (pps ppx_expect)))
```

Comment:
Can you also add this as a test dependency of the `pure-html` package in the `dune-project` file? As that will be needed to install correctly for CI. Example of how to do that: https://github.com/yawaramin/ocaml-decimal/blob/ddef44e4c625ee507aab7bddb6350bd092cc475e/dune-project#L29

Remember to run `dune build` once after updating the `dune-project` file so that it can regenerate the opam files. Thanks!

Reply:
Also `ppx_expect`. Although removing `fmt` from the `(libraries)` stanza compiles, because it already indirectly depends on it: `dream_html -> dream -> fmt`.

Changes:
- Separated dream-html and pure-html tests into two libraries (otherwise both would try to run for pure-html runtest).
- Bumped dune language version to 2.8, which is the first version that allows `(library (package ...))`.